docs/architecture/adr-083-out-of-band-sync.md

```diff
@@ -0,0 +1,73 @@
+# ADR 083: Supporting out of band state sync
+
+## Changelog
+
+- 2022-10-31: Initial Draft (@cmwaters)
+
+## Status
+
+Accepted
+
+## Context
+
+Whenever Tendermint begins, it initiates a handshake with the application via `Info` to gauge version compatibility and recieve the last height and app hash of the application. Tendermint then begins a replay protocol which aims to sync the heights of Tendermint's state and block store and the application's current height. 
+
+When initially designed, one invariant was that the applications current height should never exceed
+Tendermint's height. The protocol would error and the node would shut down if this were to happen. It seemed
+appropriate at the time that Tendermint which would provide blocks either through block sync or consensus
+would always be ahead. However, in light of state sync and other protocols for bootstrapping an application to 
+a particular height, this invariant is no longer necessary.
```

Comment:
I am not 100% I understand how can the application be ahead of Tendermint? If it takes a snapshot before Tendermint progresses? I am not 100% clear how can this happen.

Reply:
I can see this happening during recovery (WAL reply), I can't see this happening during regular operation. An example of situation when this might happen could be valuable.

Reply:
Here are some use cases where this might happen:
- I am an operator with multiple machines. I want to start a new node. Instead of relying on the network to find a snapshot, I take one from one of my nodes that I trust and add it to the new machine, rebuild state and then start an "empty" tendermint process
- I am an operator and I come across an app hash mismatch because of some non-determinism (or any general corruption of state). As I keep snapshots, I can nuke application and tendermint state and rebuild everything from the height of the previous snapshot.